tests/bundle/Functional/SearchView/SearchViewTest.php

```diff
@@ -74,6 +74,40 @@ public function testSimpleXmlContentQuery(string $xmlQueryBody, int $expectedCou
         self::assertEquals($expectedCount, $this->getQueryResultsCount('xml', $body));
     }
 
+    public function testLocationsByIdQueryContainsCurrentVersionObject(): void
+    {
+        $body = <<< JSON
+{
+    "ViewInput": {
+        "identifier": "locations-by-id",
+        "public": "false",
+        "LocationQuery": {
+            "Filter": {
+                "LocationIdCriterion": "2"
+            },
+            "limit": "10",
+            "offset": "0"
+        }
+    }
+}
+JSON;
+
+        $request = $this->createHttpRequest(
+            'POST',
+            '/api/ezp/v2/views',
+            'ViewInput+json; version=1.1',
+            'View+json',
+            $body
+        );
+
+        $response = $this->sendHttpRequest($request);
+        $jsonResponse = json_decode($response->getBody()->getContents());
```

Comment:
```suggestion
        $jsonResponse = json_decode($response->getBody()->getContents(), true);
```

This will change the result to an associative array, which is preferred to `stdClass` usage.

Reply:
Changed, but why is it preferred? All the methods use `stdClass`, for instance here: https://github.com/ezsystems/ezplatform-rest/blob/1.3/tests/bundle/Functional/SearchView/SearchViewTestCase.php#L37 so there is incosistency.

Reply:
Using `stdClass` is generally frowned upon: when using objects, a specific object class is usually expected, because using objects with declared properties is better performance-wise (since PHP 5.4). Declared properties give ~3/4 lower memory usage.

See https://gist.github.com/nikic/5015323 for details.

Therefore, using arrays more explicitly states that the data structure in question is uncertain and cannot be pre-declared.

We do not have any methods nor declared fields, so array is the correct data structure. Technically objects take a little more memory, but it's negligible.

**If it's an already established practice, I'm ok with this remaining as an object. I'd like @alongosz to take an opinion here. Language-wise there is little difference.**

Reply:
> **If it's an already established practice, I'm ok with this remaining as an object. I'd like @alongosz to take an opinion here. Language-wise there is little difference.**

AFAIR all the functional tests are written like this. @Steveb-p If you feel like we should move in the direction of arrays, we can do it as an evolution, starting here. If you don't feel strong about it, it can stay for the consistency's sake in this case. No strong opinion from my side here.

Reply:
Then let's revert it, especially since @barw4 has a different opinion here from mine :smiling_face_with_tear: